fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SizeStatisticsCollector.java

```diff
@@ -334,47 +352,23 @@ public static SizeStatisticsCollector ofIndex(@Nonnull FDBRecordStore store, @No
      *
      * @param store a store with the given index
      * @param index the index to collect statistics on key and value sizes
+     *
      * @return a statistics collector of the given index
      */
     @Nonnull
     public static SizeStatisticsCollector ofIndex(@Nonnull FDBRecordStore store, @Nonnull Index index) {
-        return new SizeStatisticsCollector(store.indexSubspace(index));
+        return new SizeStatisticsCollector(new SubspaceProviderBySubspace(store.indexSubspace(index)));
     }
 
     /**
      * Create a statistics collector of all keys used by index within a given {@link Subspace}.
      *
      * @param subspace the subspace to collect statistics on key and value sizes
-     * @return a statistics collector of the given subspace
-     */
-    @Nonnull
-    public static SizeStatisticsCollector ofSubspace(@Nonnull Subspace subspace) {
-        return new SizeStatisticsCollector(subspace);
-    }
-
-    /**
-     * Create a statistics collector of all keys used by index within a given {@link SubspaceProvider}'s subspace.
-     * If the implementation of {@link SubspaceProvider#getSubspace() getSubspace()} is blocking for the
-     * given <code>SubspaceProvide</code>, then this method will also be blocking.
      *
-     * @param subspaceProvider the provider of the subspace to collect statistics on key and value sizes
      * @return a statistics collector of the given subspace
      */
     @Nonnull
-    public static SizeStatisticsCollector ofSubspaceProvider(@Nonnull SubspaceProvider subspaceProvider) {
-        return new SizeStatisticsCollector(subspaceProvider.getSubspace());
```

Comment:
It seems like the `ofSubspaceProvider` static initializer could come back.

If we make `SubspaceProvider` internal, maybe it's not necessary, though.

Reply:
The method was not used.

Reply:
Hm, well, it was kind of intended to be the front-end for consumers of the library, so I don't think the fact that it was unused within the Record Layer necessarily means it's unused globally. However, I can get behind removing it as part of making `SubspaceProvider` internal.